app/Http/Controllers/DepartmentController.php

```diff
@@ -15,26 +25,60 @@ class DepartmentController extends Controller
     public function index()
     {
         $departments = Department::all();
+        $sectors = Sector::all();
 
         return view('departments.index', [
-            'departments'     => $departments
+            'departments'     => $departments,
+            'sectors'         => $sectors
         ]);
     }
 
     /**
      * Show the form for creating a new resource.
      */
-    public function create()
+    public function create(Request $request) : View
     {
-        return view('departments.create');
+        if($request->user()->isAdmin())
+        {
+            $sectors = Sector::all();
+            return view('departments.create', [
+                'sectors'   => $sectors,
+            ]);
+        }
+        else
+        {
+            abort(401);
+        }
     }
 
     /**
      * Store a newly created resource in storage.
      */
-    public function store(Request $request)
+    public function store(Request $request) : RedirectResponse
     {
-        //
+        // Validate the incoming request data
+        $validated = $request->validate([
+            'name' => ['required','string','max:255'], // required string, max len 255
+            'description' => ['nullable','string','max:255'],  // optional string, max len 255
+            'sector_id' => ['required','integer','exists:sectors,id']     // Ensure 'sector' is a valid integer and exists in the sectors table
+        ]);
+
+        // Programmatically determine the department ID of this new department
+        $lastDepartmentId = Department::latest()->first()->id ?? -1; //if no other department exists, start at -1+1 (0)
+        $newDepartmentId = $lastDepartmentId + 1;
+        $validated['id'] = $newDepartmentId; 
```

Comment:
This isn't needed. primary key ID's with laravel are automatically incrementally generated for us. We dont need to do this. But I see what you were doing.

Reply:
resolved in 59ff09f7743b739de37cce6a45b1e3a7053d17fb